source/fundamentals/transactions.txt

```diff
@@ -0,0 +1,159 @@
+grouping of related read or write operations that you want to run
+sequentially. Sessions enable :manual:`causal consistency </core/read-isolation-consistency-recency/#causal-consistency>`
+for a group of operations or allow you to execute operations in an :website:`ACID
+transaction </basics/acid-transactions>`. MongoDB guarantees that the data
+involved in your transaction operations remains consistent, even if the operations
+encounter unexpected errors.
+
+With the {+driver-short+}, you can create a new session from a
+``Client`` instance as a ``Session`` type.
+
+.. warning::
+
+   You should only use ``Session`` with the ``Client`` (or associated
+   ``Database`` or ``Collection``) that created the ``Session``. Using a
+   ``Session`` with a different ``Client`` will result in operation
+   errors.
+
+.. warning::
+   
+   Implementations of ``Session`` are not safe for concurrent use by multiple `goroutines
+   <https://www.golang-book.com/books/intro/10>`__. Additionally, you
+   should not use custom implementations of the ``Session`` interface in
+   production.
+
+Methods
+-------
+
+After you start a session using the ``StartSession()`` method, you can modify
+the session state using the method set in the ``Session`` interface. The
+following table describes these methods:
+
+.. list-table::
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``StartTransaction()``
+     - | Starts a new transaction, configured with the given options, on
+         this session. Returns an error if there is already
+         a transaction in progress for the session. For more
+         information, see the :manual:`Manual entry </reference/method/Session.startTransaction/>`.
+       |
+       | **Parameter**: ``TransactionOptions``
+       | **Return Type**: ``error``
+
+   * - ``AbortTransaction()``
+     - | Aborts the active transaction for this session. Returns an
+         error if there is no active transaction for the session or the
+         transaction has been committed or aborted. For more
+         information, see the :manual:`Manual entry </reference/method/Session.abortTransaction/>`.
+       |
+       | **Parameter**: ``Context``
+       | **Return Type**: ``error``
+
+   * - ``CommitTransaction()``
+     - | Commits the active transaction for this session. Returns an
+         error if there is no active transaction for the session or the
+         transaction has been aborted. For more
+         information, see the :manual:`Manual entry </reference/method/Session.commitTransaction/>`.
+       |
+       | **Parameter**: ``Context``
+       | **Return Type**: ``error``
+
+   * - ``WithTransaction()``
+     - | Starts a transaction on this session and runs the ``fn``
+         callback. For more
+         information, see the :manual:`Manual entry </reference/method/Session.withTransaction/>`.
+       |
+       | **Parameter**: ``Context``, ``fn func(sessCtx SessionContext)``, ``TransactionOptions``
```

Comment:
Optional: We're changing this function signature in the next release to use `ctx` instead of `sessCtx` to match the Go Driver example guidelines (see [GODRIVER-2630](https://jira.mongodb.org/browse/GODRIVER-2630)). Consider using `ctx` instead of `sessCtx` to reduce confusion.

```suggestion
       | **Parameter**: ``Context``, ``fn func(ctx SessionContext)``, ``TransactionOptions``
```